spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java

```diff
@@ -138,4 +140,10 @@ public KafkaAdmin kafkaAdmin() {
 		return kafkaAdmin;
 	}
 
+	@Bean
+	@ConditionalOnMissingBean(AdminClient.class)
+	public AdminClient adminClient() {
```

Comment:
This is a separate feature that shouldn't be part of this change. Can you please create a separate PR for this?